server/groups/class.lua

```diff
@@ -85,6 +88,15 @@ function CGroup:set(player, grade)
     if currentGrade then
         if currentGrade == grade then return end
         self:remove(player, currentGrade)
+    else
+        if self.type and self.unique_type == 1 then
+            for _ in pairs(player.private.groups) do
```

Comment:
Weird to use `_` when using the variable. Would also be good to have an exported function or method to check if a group can be added.